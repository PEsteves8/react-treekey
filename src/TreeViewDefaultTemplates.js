import React from "react";
import styles from "./defaultTreeStyles";

export let defaultToggle = (isExpanded) => {
  const svgStyles = {
    fill: "white",
    strokeWidth: 0,
    transformOrigin: "center",
    transform: `rotate(${isExpanded ? "90deg" : "0"})`,
  };

  let { height, width, points } = styles.nodeToggleSvg;

  return (
    <svg height={height} width={width}>
      <polygon points={points} style={svgStyles} />
    </svg>
  );
};

export let defaultHeader = (node) => {
  return <span style={{ paddingLeft: "7px" }}>{node.name}</span>;
};
