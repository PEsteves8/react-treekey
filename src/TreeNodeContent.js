import React from 'react';

import { defaultHeader } from './TreeViewDefaultTemplates';

export default class TreeListItemContent extends React.Component {

    renderNodeContent() {
        let { template, node } = this.props;
        return template ? template(node) : defaultHeader(node);
    }

    render() {
        return this.renderNodeContent();
    }
}