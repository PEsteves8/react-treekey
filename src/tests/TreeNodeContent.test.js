import React from "react";
import TreeNodeContent from "../TreeNodeContent";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import * as templates from "../TreeViewDefaultTemplates";

const mockNode = {
  name: "some node",
};

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("<TreeNode>", () => {
  test("should match snapshot calling correct template", () => {
    let template = jest.fn();
    template.mockImplementation(() => <span>some node</span>);

    act(() => {
      render(
        <TreeNodeContent template={template} node={mockNode}></TreeNodeContent>,
        container
      );
    });

    expect(container.innerHTML).toMatchSnapshot();
    expect(template).toHaveBeenCalledTimes(1);
    expect(template).toHaveBeenCalledWith(mockNode);
  });

  test("calls default template if no template is given", () => {
    const spy = jest.spyOn(templates, "defaultHeader");

    act(() => {
      render(<TreeNodeContent node={mockNode}></TreeNodeContent>, container);
    });

    expect(spy).toHaveBeenCalledTimes(1);
    expect(spy).toHaveBeenCalledWith(mockNode);
    spy.mockRestore();
  });
});
