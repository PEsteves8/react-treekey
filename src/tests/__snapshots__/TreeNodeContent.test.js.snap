// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreeNode> should match snapshot calling correct template 1`] = `"<span>some node</span>"`;
